### PR TITLE
Introduced foldLeftBB

### PR DESCRIPTION
**Pull request by pchlupacek**
@mpilquist as discussed on gitter

### CODE REVIEW

**Comment by pchlupacek** [comment]
Also I sort of thought that perhaps there can be generalised fold over View that will be private, and then we would be able to define foreachV and foldBB with it, but not sure if that is worth of effort.

**Comment by pchlupacek** [comment]
I think the build failure is unrelated?

**Comment by pchlupacek** [comment]
Also I am still in sort of inclining to have unsafe variant of foldLeft (i.e. foldLeftUnsafeArray and foldRightUnsafeArray), that instead of buffer will have Array[Byte], Int, Int as parameters. In case of AtEmpty this will be filled with Array.empty, 0,0, in case of the AtBuffer, there will be copy to array required. But I still believe it is worth having one. What do you think?

File: core/shared/src/main/scala/scodec/bits/ByteVector.scala [hunk 1]
```diff
@@ -319,6 +319,32 @@ sealed abstract class ByteVector extends BitwiseOperations[ByteVector, Long] wit
     reverse.foldLeft(z)((tl,h) => f(h,tl))
 
   /**
+    * Applies a binary operator to a start value and all segments(views) of this ByteVector expressed as read-only ByteBuffer, going left to right.
+    * @param z    Starting value
+    * @param f    operator to apply
+    */
```
Comment (mpilquist) [inline core/shared/src/main/scala/scodec/bits/ByteVector.scala]: Add `@group collection`  here and on foldRightBB -- this project uses ScalaDoc groups so every member needs a group annotation.

**Comment by mpilquist** [comment]
Looks good -- 1 minor ScalaDoc comment and then we can merge.

I'm not sure about `foldLeftUnsafeArray` -- I think the byte buffer variant would be indistinguishable from the array variant in all but the microest of micro benchmarks. :)

**Comment by mpilquist** [comment]
I'm going to merge this and then change the scaladoc manually so I can get a build out for 2.12.0-RC1. Thanks.

**Comment by pchlupacek** [comment]
sorry, I didn't follow it in time

**Comment by mpilquist** [comment]
No worries -- I thought I could get 1.1.1 built but we're blocked on a scalacheck and scalatest release for RC1. Maybe tomorrow. :)

**Pull request merged**